vello_encoding/src/ramp_cache.rs

```diff
@@ -35,13 +36,14 @@ impl RampCache {
     }
 
     pub(crate) fn add(&mut self, stops: &[ColorStop]) -> u32 {
-        if let Some(entry) = self.map.get_mut(stops) {
+        if let Some(entry) = self.map.get_mut(&CacheKey(ColorStops(stops.into()))) {
             entry.1 = self.epoch;
             entry.0
         } else if self.map.len() < RETAINED_COUNT {
             let id = (self.data.len() / N_SAMPLES) as u32;
             self.data.extend(make_ramp(stops));
-            self.map.insert(stops.into(), (id, self.epoch));
+            self.map
+                .insert(CacheKey(ColorStops(stops.into())), (id, self.epoch));
```

Comment:
I'm not thrilled about this change to `ColorStops(stops.into())` rather than perhaps going and doing some better impls for `From` or something in Peniko.

Reply:
https://github.com/linebender/peniko/pull/86 allows for `CacheKey(ColorStops::from(stops))` / `CacheKey(stops.into())`.

Reply:
But really this copying of the color stops just to serve as key seems suboptimal since what we could probably do this with the `Range<usize>` that we have available in the resolver code. But that's not for this PR.

Reply:
We can't currently do that ... I just tried, but we can't rely on data in the `Resources::color_stops` being uniqued, so we can't use the range of it as a key.